packages/backend/src/controllers/authentication.ts

```diff
@@ -499,28 +532,174 @@ export const isOktaPassportStrategyAvailableToUse = !!(
     lightdashConfig.auth.okta.oktaDomain
 );
 
-export const azureAdPassportStrategy = !(
-    lightdashConfig.auth.azuread.oauth2ClientId &&
-    lightdashConfig.auth.azuread.oauth2ClientSecret &&
-    lightdashConfig.auth.azuread.oauth2TenantId
-)
-    ? undefined
-    : new OpenIDConnectStrategy(
-          {
-              issuer: `https://login.microsoftonline.com/${lightdashConfig.auth.azuread.oauth2TenantId}/v2.0`,
-              authorizationURL: `https://login.microsoftonline.com/${lightdashConfig.auth.azuread.oauth2TenantId}/oauth2/v2.0/authorize`,
-              tokenURL: `https://login.microsoftonline.com/${lightdashConfig.auth.azuread.oauth2TenantId}/oauth2/v2.0/token`,
-              userInfoURL: 'https://graph.microsoft.com/oidc/userinfo',
-              clientID: lightdashConfig.auth.azuread.oauth2ClientId,
-              clientSecret: lightdashConfig.auth.azuread.oauth2ClientSecret,
-              callbackURL: new URL(
-                  `/api/v1${lightdashConfig.auth.azuread.callbackPath}`,
-                  lightdashConfig.siteUrl,
-              ).href,
-              passReqToCallback: true,
-          },
-          genericOidcHandler(OpenIdIdentityIssuerType.AZUREAD),
-      );
+/**
+ * Loads a JWK file from a given path
+ */
+const loadJWKFromPath = async (
+    filePath: string,
+): Promise<Record<string, unknown>> => {
+    try {
+        const file = await fs.readFile(filePath, 'utf-8');
+        return JSON.parse(file);
+    } catch (err) {
+        throw new Error(
+            `Failed to load JWK from file at '${filePath}' - may be missing or not valid JSON`,
+            err,
+        );
+    }
+};
+
+/**
+ * Loads a JWK from a JSON-encoded string, likely provided as an environment variable.
+ */
+const loadJWKFromConfig = (jwkConfigValue: string): Record<string, unknown> => {
+    try {
+        return JSON.parse(jwkConfigValue);
+    } catch (err) {
+        throw new Error(
+            `Failed to parse JWK from configuration - must be valid JSON`,
+            err,
+        );
+    }
+};
```

Comment:
These functions are not used atm.

Reply:
Good catch, I refactored the configuration process late at night - will clean this up.